runbotics/runbotics-scheduler/src/websocket/bot/bot-lifecycle.service.ts

```diff
@@ -1,62 +1,80 @@
+                    return null;
                 });
+            
+        const newProcessInstance: IProcessInstance = {
+            ...completeProcessInstance,
+            status: ProcessInstanceStatus.ERRORED, 
+            error: 'Bot has been shut down',
+            updated: new Date().toString(),
+        };
 
-            this.handleInterruptedProcessInstanceEvent(bot, processInstance.id);
-        });
+        this.updateInterruptedProcessInstance(newProcessInstance, bot.installationId);            
+        this.handleProcessInstanceEventInterruption(disconnectedInstance, bot);
     }
 
-    private async handleInterruptedProcessInstanceEvent (bot: IBot, processInstanceId: string): Promise<void> {
-        const activeProcessInstanceEvents = this.processInstanceEventService.findActiveByProcessInstanceId(processInstanceId);
-        const newProcessInstanceEvent = {
-            ...activeProcessInstanceEvents,
-            status: ProcessInstanceEventStatus.ERRORED,
-            error: 'Bot has been shut down',
-        };
+    private async handleProcessInstanceEventInterruption (processInstance: IProcessInstance, bot: IBot): Promise<void> {
+        const activeEvents: IProcessInstanceEvent[] = 
+            await this.processInstanceEventService
+                .findActiveByProcessInstanceId(processInstance.id);
+                
+        if(!activeEvents) return;
+
+        await activeEvents.forEach(async event => {
+            const newProcessInstanceEvent: IProcessInstanceEvent = {
+                ...event, 
+                status: ProcessInstanceEventStatus.ERRORED, 
+                error: 'Bot has been shut down',
+                finished: processInstance.updated,
+                processInstance,
+            };
+            await this.updateInterruptedProcessInstanceEvent(newProcessInstanceEvent, bot);
+        });
+    }
 
-        this.botProcessEventService.updateProcessInstanceEvent(newProcessInstanceEvent, bot);
+    private async updateInterruptedProcessInstance(processInstance: IProcessInstance, installationId: string) {
+        this.logger.log(`=> Updating process-instance (${processInstance.id}) after bot disconnection | status: ${processInstance.status}`);
+        await this.botProcessService.updateProcessInstance(installationId, processInstance);
+        this.logger.log(`<= Success process-instance (${processInstance.id}) updated | status: ${processInstance.status}`);
+    }
+    
+    private async updateInterruptedProcessInstanceEvent(processInstanceEvent: IProcessInstanceEvent, bot: IBot) {
+        this.logger.log(`=> Updating process-instance-event (${processInstanceEvent.id}) | status: ${processInstanceEvent.status}`);
+        await this.botProcessEventService.updateProcessInstanceEvent(processInstanceEvent, bot);
+        this.logger.log(`<= Success process-instance-event (${processInstanceEvent.id}) updated | status: ${processInstanceEvent.status}`);
```

Comment:
remove arrows in log messages + add info about interruption

Reply:
done